fractal.sdk/src/main/java/com/yanchware/fractal/sdk/domain/entities/ComponentLink.java

```diff
@@ -54,7 +58,17 @@ public ComponentLinkBuilder withComponentId(String componentId) {
         }
 
         public ComponentLinkBuilder withSettings(Map<String, Object> settings) {
-            componentLink.setSettings(settings);
+            componentLink.setSettings(new HashMap<>(settings));
+            return builder;
+        }
+
+        public ComponentLinkBuilder withRoleName(String roleName) {
+            if (componentLink.settings.containsKey(ROLE_NAME_PARAM_KEY)) {
+                componentLink.settings.replace(ROLE_NAME_PARAM_KEY, roleName);
```

Comment:
This code looks really strange.
We use `componentLink.settings` and `componentLink.getSettings()` instead of using one of them consistently.
Moreover, why do we check and put? [Put already replace the value if exists](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/util/HashMap.html#put(java.lang.Object,java.lang.Object)).